src/ios/GMImagePicker/GMGridViewController.m

```diff
@@ -469,69 +469,73 @@ - (BOOL)collectionView:(UICollectionView *)collectionView shouldSelectItemAtInde
         }];
         
         
-            
-        [ self.imageManager requestImageForAsset:asset targetSize:PHImageManagerMaximumSize contentMode:PHImageContentModeDefault options:ph_options resultHandler:^(UIImage *result, NSDictionary *info) {
-            
-            //dispatch_async(dispatch_get_main_queue(), ^{
-            
-            GMGridViewCell *cell = (GMGridViewCell *)[collectionView cellForItemAtIndexPath:indexPath];
-            
-            if ( cell ) {
-                [cell hide_progress];
-                [cell show_fetching];
-            }
-            
-            fetch_item.be_progressed = false;
-            fetch_item.be_finished = true;
-            
-            //asset.image_fullsize = result;
-            
-            NSString * filePath;
-            do {
-                filePath = [NSString stringWithFormat:@"%@/%@%03d.%@", docsPath, CDV_PHOTO_PREFIX, docCount++, @"jpg"];
-            } while ([fileMgr fileExistsAtPath:filePath]);
-            
-            fetch_item.be_saving_img = true;
-            
-            dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-                
+        if (asset.mediaType == PHAssetMediaTypeImage) {
+            [ self.imageManager requestImageForAsset:asset targetSize:PHImageManagerMaximumSize contentMode:PHImageContentModeDefault options:ph_options resultHandler:^(UIImage *result, NSDictionary *info) {
                 
-                // @BVL: Added orientation-fix to correctly display the returned result
+                GMGridViewCell *cell = (GMGridViewCell *)[collectionView cellForItemAtIndexPath:indexPath];
                 
-//              if ( ![ UIImageJPEGRepresentation(result, 1.0f ) writeToFile:filePath atomically:YES ] ) {
-//                  return;
-//              }
+                if (cell) {
+                    [cell hide_progress];
+                    [cell show_fetching];
+                }
                 
-                NSLog(@"original orientation: %ld",(UIImageOrientation)result.imageOrientation);
+                fetch_item.be_progressed = false;
+                fetch_item.be_finished = true;
                 
-                UIImage *imageToDisplay = result.fixOrientation; //  UIImage+fixOrientation extension
                 
-          		NSLog(@"corrected orientation: %ld",(UIImageOrientation)imageToDisplay.imageOrientation);
-
-                if ( ![ UIImageJPEGRepresentation(imageToDisplay, 1.0f ) writeToFile:filePath atomically:YES ] ) {
-                    return;
-                }
+                NSString * filePath;
+                do {
+                    filePath = [NSString stringWithFormat:@"%@/%@%03d.%@", docsPath, CDV_PHOTO_PREFIX, docCount++, @"jpg"];
+                } while ([fileMgr fileExistsAtPath:filePath]);
```

Comment:
Can you explain what this code is doing? It's not immediately clear to me. It looks like it's iterating through the count of documents and setting the file path until it doesn't find one... just not sure why?